src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java

```diff
@@ -27,12 +32,18 @@
 
 public class HologramProjector extends SlimefunItem {
 
-    private static final String OFFSET_PARAMETER = "offset";
+    private static final String X_OFFSET_PARAMETER = "x_offset";
+    private static final String Y_OFFSET_PARAMETER = "y_offset";
+    private static final String Z_OFFSET_PARAMETER = "z_offset";
+    private static final int[] BORDER = {1, 3, 5, 7, 8, 10, 12, 14, 16, 17};
+
+    private final DoubleRangeSetting maxDistance = new DoubleRangeSetting("max-horizontal-offset", 0.0, 5.0, 10.0);
```

Comment:
Let's rather declare this with the base type.
```suggestion
    private final ItemSetting<Double> maxDistance = new DoubleRangeSetting("max-horizontal-offset", 0.0, 5.0, 10.0);
```